src/Support/Str/Friendly.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * This file is part of the Phalcon.
+ *
+ * (c) Phalcon Team <team@phalcon.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Phalcon\Support\Str;
+
+use Phalcon\Support\Exception;
+use Phalcon\Support\Str\Traits\LowerTrait;
+
+use function is_array;
+use function is_string;
+use function preg_replace;
+use function str_replace;
+use function trim;
+
+/**
+ * Class Friendly
+ *
+ * @package Phalcon\Support\Str
+ */
+class Friendly
+{
+    use LowerTrait;
+
+    /**
+     * Changes a text to a URL friendly one
+     *
+     * @param string     $text
+     * @param string     $separator
+     * @param bool       $lowercase
+     * @param mixed|null $replace
+     *
+     * @return string
+     * @throws Exception
+     */
+    public function __invoke(
+        string $text,
+        string $separator = '-',
+        bool $lowercase = true,
+        $replace = null
+    ): string {
+        if (null !== $replace) {
+            $replace = $this->checkParameters($replace);
+            $text    = str_replace($replace, ' ', $text);
+        }
+
+        $friendly = preg_replace(
+            '/[^a-zA-Z0-9\\/_|+ -]/',
+            '',
+            $text
+        );
+
+        if ($lowercase) {
+            $friendly = $this->toLower((string) $friendly);
```

Comment:
Since text is checked as string in parameters I think it's safe to handle the output of preg_replace as string and casting isn't needed.

Reply:
Done